File: src/main/java/qmech/mod/network/ClientProxy.java
package qmech.mod.network;

import cpw.mods.fml.client.registry.ClientRegistry;
import qmech.lib.tileentity.render.CustomRendererBase;

import java.util.Map;

public class ClientProxy extends CommonProxy {

    public void preInit() {
        this.registerRenderers();
    }

    public void registerRenderers() {
        for (Map.Entry binding : this.renderers.entrySet())
            ClientRegistry.bindTileEntitySpecialRenderer((Class) binding.getKey(), (CustomRendererBase) binding.getValue());
    }

}

File: src/main/scala/geek/lawsof/physics/network/proxy/ClientProxy.scala
package geek.lawsof.physics.network.proxy

import cpw.mods.fml.client.registry.ClientRegistry
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity

/**
 * Created by anshuman on 28-05-2014.
 */
class ClientProxy extends ModProxy {
  override def preInit(): Unit = {}

  override def init(): Unit = {}

  override def registerRenderer(te: Class[_ <: TileEntity], renderer: TileEntitySpecialRenderer): Unit = ClientRegistry.bindTileEntitySpecialRenderer(te, renderer)
}
